pkg/controller/installation/products/launcher/reconciler.go

```diff
@@ -0,0 +1,190 @@
+	if err != nil {
+		return v1alpha1.PhaseFailed, pkgerr.Wrap(err, " failed to reconcile custom resource for launcher")
+	}
+
+	phase, err = r.reconcileOauthClient(ctx, inst, serverClient)
+	if err != nil {
+		return v1alpha1.PhaseFailed, pkgerr.Wrap(err, " failed to reconcile oauthclient for launcher")
+	}
+
+	r.logger.Debug("End of reconcile Phase: ", phase)
+
+	// if we get to the end and no phase set then the reconcile is completed
+	if phase == v1alpha1.PhaseNone {
+		return v1alpha1.PhaseCompleted, nil
+	}
+
+	return phase, nil
+}
+
+func (r *Reconciler) reconcileNamespace(ctx context.Context, inst *v1alpha1.Installation, serverClient pkgclient.Client) (v1alpha1.StatusPhase, error) {
+	nsr := resources.NewNamespaceReconciler(serverClient, r.logger)
+	ns := &v1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: r.Config.GetNamespace(),
+		},
+	}
+
+	// Reconcile namespace
+	ns, err := nsr.Reconcile(ctx, ns, inst)
+	if err != nil {
+		return v1alpha1.PhaseFailed, pkgerr.Wrap(err, "failed to reconcile fuse namespace "+r.Config.GetNamespace())
+	}
+
+	if ns.Status.Phase == v1.NamespaceTerminating {
+		r.logger.Debugf("namespace %s is terminating, maintaining phase to try again on next reconcile", r.Config.GetNamespace())
+		return v1alpha1.PhaseAwaitingNS, nil
+	}
+
+	if ns.Status.Phase != v1.NamespaceActive {
+		return v1alpha1.PhaseAwaitingNS, nil
+	}
+
+	// all good return no status when ready
+	r.logger.Debug("namespace is ready")
+	return v1alpha1.PhaseNone, nil
+}
+
+func (r *Reconciler) reconcileSubscription(ctx context.Context, serverClient pkgclient.Client) (v1alpha1.StatusPhase, error) {
+	// TODO Create subscription (This will create the launcher operator)
+	logrus.Debugf("creating subscription %s from channel %s in namespace: %s", defaultSubscriptionName, "integreatly", r.Config.GetNamespace())
+	err := r.mpm.CreateSubscription(
+		marketplace.GetOperatorSources().Integreatly,
+		r.Config.GetNamespace(),
+		defaultSubscriptionName,
+		marketplace.IntegreatlyChannel,
+		[]string{r.Config.GetNamespace()},
+		coreosv1alpha1.ApprovalAutomatic)
+	if err != nil && !k8serr.IsAlreadyExists(err) {
+		return v1alpha1.PhaseFailed, pkgerr.Wrap(err, fmt.Sprintf("could not create subscription in namespace: %s", r.Config.GetNamespace()))
+	}
+
+	//return v1alpha1.PhaseAwaitingOperator, nil
+	return r.handleAwaitingOperator(ctx, serverClient)
+}
+
+func (r *Reconciler) handleAwaitingOperator(ctx context.Context, client pkgclient.Client) (v1alpha1.StatusPhase, error) {
+	r.logger.Infof("checking installplan is created for subscription %s in namespace: %s", defaultSubscriptionName, r.Config.GetNamespace())
+	ip, sub, err := r.mpm.GetSubscriptionInstallPlan(defaultSubscriptionName, r.Config.GetNamespace())
+	if err != nil {
+		if k8serr.IsNotFound(err) {
+			if sub != nil {
+				logrus.Infof("time since created %v", time.Now().Sub(sub.CreationTimestamp.Time).Seconds())
+			}
+			// should be triggered ever 3 minuets if no install plan found
```

Comment:
This should probably be uncommented or removed

Reply:
This will be removed as I don't think we'd need this in Launcher